### PR TITLE
Add proxy support

### PR DESCRIPTION
**Pull request by ecordell**

### CODE REVIEW

**Comment by ecordell** [comment]
@josephschorr

File: websocket.go [hunk 1]
```diff
@@ -17,26 +18,37 @@ type websocketPeer struct {
...
 		Subprotocols:    []string{protocol},
 		TLSClientConfig: tlscfg,
 	}
+	if proxyUrl != "" {
+		parsedProxyUrl, err := url_.Parse(proxyUrl)
+		if err != nil {
+			return nil, err
+		}
+		dialer.Proxy = http.ProxyURL(&url_.URL{
+			Scheme: "https",
```
Comment (josephschorr) [inline websocket.go]: Could this be non-https?

**Comment by jakedt** [comment]
LGTM

**Comment by ecordell** [comment]
@jakedt no write access

**Pull request merged**